src/hooks/useLinkedServiceAccounts.ts

```diff
@@ -0,0 +1,16 @@
+import { useQuery } from '@tanstack/react-query';
+import {
+  getLinkedServiceAccounts,
+  isLinkableSecret,
+} from '~/components/Secrets/utils/service-account-utils';
+import { FIVE_MINUTES } from '~/consts/time';
+import { SecretKind } from '~/types';
+
+export const useLinkedServiceAccounts = (secret: SecretKind | undefined) => {
+  return useQuery({
+    queryKey: ['linked-service-accounts', secret?.metadata?.namespace, secret?.metadata?.name],
+    queryFn: () => getLinkedServiceAccounts(secret),
+    enabled: isLinkableSecret(secret) && !!secret?.metadata?.namespace && !!secret?.metadata?.name,
+    staleTime: FIVE_MINUTES,
+  });
```

Comment:
use `useK8sWatchResource`, For K8s resource we dont need to create custom fetching utils